pkg/api/convertertoapi.go

```diff
@@ -598,12 +604,19 @@ func convertVLabsOrchestratorProfile(vp *vlabs.Properties, api *OrchestratorProf
 			convertVLabsKubernetesConfig(vlabscs.KubernetesConfig, api.KubernetesConfig)
 		}
 		setVlabsKubernetesDefaults(vp, api)
+
+		//HACK : this validation should be done as part of the main validation, but deploy does it only after loading the container.
```

Comment:
Super nit: This tools will pick up on this as a TODO item. Better for future tracking purposes
```suggestion
		// TODO (hack): this validation should be done as part of the main validation, but deploy does it only after loading the container.
```